AnkiDroid/src/main/java/com/ichi2/anki/dialogs/EmptyCardsDialogFragment.kt

```diff
@@ -0,0 +1,173 @@
+                        )
+                    }
+                }
+                setNegativeButton(R.string.dialog_cancel) { _, _ ->
+                    Timber.i("Empty cards dialog cancelled")
+                }
+                setView(dialogView)
+            }.also {
+                // the initial start state is a loading state as we are looking for the empty cards,
+                // so there's no "action" for ok just yet
+                it.positiveButton.isEnabled = false
+            }
+    }
+
+    private fun bindToState() {
+        lifecycleScope.launch {
+            repeatOnLifecycle(Lifecycle.State.STARTED) {
+                viewModel.uiState.collect { state ->
+                    when (state) {
+                        is SearchingForEmptyCards -> {
+                            loadingMessage?.text = getString(R.string.emtpy_cards_finding)
+                            loadingContainer?.isVisible = true
+                            emptyCardsResultsContainer?.isVisible = false
+                            (dialog as? AlertDialog)?.positiveButton?.apply {
+                                isEnabled = false
+                                text = getString(R.string.dialog_ok)
+                            }
+                        }
+
+                        is EmptyCardsSearchResult -> {
+                            loadingContainer?.isVisible = false
+                            val emptyCards = state.emptyCardsReport.emptyCids()
+                            if (emptyCards.isEmpty()) {
+                                emptyCardsResultsMessage?.text = TR.emptyCardsNotFound()
+                                // nothing to delete so also hide the preserve notes check box
+                                keepNotesWithNoValidCards?.isVisible = false
+                                (dialog as? AlertDialog)?.positiveButton?.text =
+                                    getString(R.string.dialog_ok)
+                            } else {
+                                keepNotesWithNoValidCards?.isVisible = true
+                                emptyCardsResultsMessage?.text =
+                                    getString(R.string.empty_cards_count, emptyCards.size)
+                                (dialog as? AlertDialog)?.positiveButton?.text =
+                                    getString(R.string.dialog_positive_delete)
+                            }
+                            (dialog as? AlertDialog)?.positiveButton?.isEnabled = true
+                            emptyCardsResultsContainer?.isVisible = true
+                        }
+
+                        is EmptyCardsSearchFailure -> {
+                            // the dialog is informational so there's nothing to do but show the
+                            // error and exit
+                            requireActivity().showSnackbar(state.throwable.toString())
+                            dismiss()
```

Comment:
```suggestion
                            AlertDialog.Builder(requireActivity()).show {
                                message(text = state.throwable.toString())
                                positiveButton(R.string.dialog_ok) {  }
                            }
                            dismissNow()
```

A snackbar is quite limited for the allowed text size, a dialog should be better I think. I used _dismissNow_ to dismiss the dialog as fast as possible.